R/tune.R

```diff
@@ -0,0 +1,182 @@
+#' values are evaluated. Conversely, a value less than one will diminish the
+#' influence of that metric. The `categories()` does not have a scaling
+#' argument while `target()` has two (`scale_low` and `scale_high`) for ranges
+#' on either side of the `target`.
+#'
+#' Here is an example that optimizes RMSE and the concordance correlation
+#' coefficient (a.k.a. "ccc"), with more emphasis on the former:
+#'
+#' \preformatted{
+#'   minimize(rmse, low = 0.10, high = 2.00, scale = 3.0),
+#'   maximize(ccc,  low = 0.00, high = 1.00) # scale defaults to 1.0
+#' }
+#'
+#' If `low`, `high`, or `target` are not specified, the observed data are used
+#' to estimate their values. For the previous example, if we were to use
+#'
+#' \preformatted{
+#'   minimize(rmse, low = 0.10, high = 2.00, scale = 3.0),
+#'   maximize(ccc)
+#' }
+#'
+#' and the concordance correlation coefficient statistics ranged from 0.21 to
+#' 0.35, the actual goals would  end up as:
+#'
+#' \preformatted{
+#'   minimize(rmse, low = 0.10, high = 2.00, scale = 3.0),
+#'   maximize(ccc,  low = 0.21, high = 0.35)
+#' }
+#'
+#' More than one variable can be used in a term as long as R can parse and
+#' execute the expression. For example, you could define the Youden's J
+#' statistic using
+#'
+#' \preformatted{
+#'   maximize(sensitivity + specificity - 1)
+#' }
+#'
+#' (although there is a function for this metric in the \pkg{yardstick} package).
+#'
+#' If the columns of the data set have missing values, their corresponding
+#' desirability will be missing. The overall desirability computation excludes
+#' missing values.
+#'
+#' We advise not referencing global values or inline functions inside of these
+#' verbs.
+#'
+#' Also note that there may be more than `n` values returned when showing the
+#' results; there may be more than one model configuration that has identical
+#' overall desirability.
+#'
+#' @seealso [d_max()], [d_overall()]
+#' @references Derringer, G. and Suich, R. (1980), Simultaneous Optimization of
+#' Several Response Variables. _Journal of Quality Technology_, 12, 214-219.
+#'
+#' Bartz-Beielstein, T. (2025). Multi-Objective Optimization and Hyperparameter
+#' Tuning With Desirability Functions. arXiv preprint arXiv:2503.23595.
+#'
+#' @examples
+#' # use pre-tuned results to demonstrate:
+#' if (rlang::is_installed("tune")) {
+#'
+#'   show_best_desirability(
+#'     tune::ames_iter_search,
+#'     maximize(rsq),
+#'     minimize(rmse, scale = 3)
+#'   )
+#'
+#'   select_best_desirability(
+#'     tune::ames_iter_search,
+#'     maximize(rsq),
+#'     minimize(rmse, scale = 3)
+#'   )
+#' }
+#'
+#' @export
+show_best_desirability <- function(x, ..., n = 5, eval_time = NULL) {
+  rlang::check_installed("tune")
+  mtr_set <- tune::.get_tune_metrics(x)
+
+  mtr <- tune::collect_metrics(x, type = "wide")
+  all_vars <- names(mtr)
+  all_vars <- all_vars[all_vars != ".config"]
+  # TODO filter on eval_time
```

Comment:
I need to do more checking for dynamic survival models and ensure that they properly use the "wide" format in `collect_metrics()`.